engine/src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -883,6 +894,18 @@ case class JesBackend(actorSystem: ActorSystem)
     lazy val jesStderrGcsPath = callGcsPath.resolve(jesLogStderrFilename(key)).toString
 
     val callContext = new CallContext(callGcsPath.toString, jesStdoutGcsPath, jesStderrGcsPath)
-    new JesCallEngineFunctions(workflowDescriptor.ioManager, callContext)
+    new JesCallEngineFunctions(workflowDescriptor.fileSystems, callContext)
+  }
+
+  override def fileSystems(options: WorkflowOptions, workflowRootPath: String): List[FileSystem] = {
+    // TODO Should GCS operations have their own execution context ?
+    implicit val executionContext = scala.concurrent.ExecutionContext.global
```

Comment:
I like the notion of GCS having its own threadpool

@kshakir Can you write this up as a ticket to investigate?